testing/test_cache.py

```diff
@@ -192,13 +192,34 @@ def test_always_fails():
             "test_a.py*",
             "test_b.py*",
         ])
-        result = testdir.runpytest("--lf", "--ff")
+        result = testdir.runpytest("--ff")
         # Test order will be failing tests firs
         result.stdout.fnmatch_lines([
             "test_b.py*",
             "test_a.py*",
         ])
 
+    def test_lastfailed_failedfirst_order(self, testdir):
+        testdir.tmpdir.join('test_a.py').write(_pytest._code.Source("""
```

Comment:
These files can be created more easily like this:

```python
    def test_lastfailed_failedfirst_order(self, testdir):
        testdir.makepyfile({
            'test_a.py': """
                def test_always_passes():
                    assert 1
            """,
            'test_b.py': """
                def test_always_fails():
                    assert 0
            """),
        })        
```

But this is just FYI really, no need to actually change it.